container.go

```diff
@@ -220,80 +236,128 @@ func (c Container) Call(function interface{}) error {
 }
 
 // Resolve takes an abstraction (reference of an interface type) and fills it with the related concrete.
-func (c Container) Resolve(abstraction interface{}) error {
-	return c.NamedResolve(abstraction, "")
+func (c *Container) Resolve(abstraction interface{}) error {
+	return c.ResolvedNamed(abstraction, "")
 }
 
-// NamedResolve takes abstraction and its name and fills it with the related concrete.
-func (c Container) NamedResolve(abstraction interface{}, name string) error {
-	receiverType := reflect.TypeOf(abstraction)
-	if receiverType == nil {
+// ResolveWithContext takes an abstraction and a context and fills it with the related concrete.
+func (c *Container) ResolveWithContext(ctx context.Context, abstraction interface{}) error {
+	err := c.RegisterSingleton(func() context.Context {
+		return ctx
+	})
+	if err != nil {
+		return err
+	}
+
+	if err := c.Resolve(abstraction); err != nil {
+		return err
+	}
+
+	contextType := reflect.TypeOf((*context.Context)(nil)).Elem()
+	delete(c.bindings, contextType)
+
+	return nil
+}
+
+// ResolveAll takes an abstraction and returns all the related concretes with their names.
+func (c *Container) ResolveAll(interfaceType reflect.Type, abstraction interface{}) error {
```

Comment:
Can you update the comment?  I'm not sure I understand the purpose of this method and the comment seems out of date.

Reply:
I've removed this.  We don't need it now but similar behavior does exist on other IoC frameworks.  The idea is to resolve all instance of a given type.  Ex) You want to get an instance of all service target implementations in one resolution.